Dockerfile

```diff
@@ -0,0 +1,53 @@
+FROM nvidia/cuda:11.3.0-cudnn8-runtime-ubuntu18.04
+
+RUN \
+    apt-get update && \
+    apt-get -y install wget \
+    aria2 \
+    rsync \
+    cmake \
+    build-essential \
+    curl \
+    git \
+    vim \
+    nvidia-cuda-toolkit
+
+# install miniconda
+RUN \
+    curl https://repo.anaconda.com/miniconda/Miniconda3-latest-Linux-x86_64.sh -o miniconda.sh && \
+    sh miniconda.sh -b -p /tools/miniconda && \
+    rm miniconda.sh && \
+    /tools/miniconda/bin/conda init && \
+    /tools/miniconda/bin/conda clean -a
+ENV PATH "$PATH:/tools/miniconda/bin"
+
+COPY . /openfold
```

Comment:
Copying the source code this early means you are going to miss cache lines and have to rebuild *a lot* of stuff everytime you edit the openfold code

Reply:
You want to move the source code copying as low as possible and stable environment building as high as possible in the dockerfile